MediaBrowser.Model/Entities/MediaStream.cs

```diff
@@ -112,107 +112,131 @@ public string DisplayTitle
+                                // keep Tags that are not already in Title
+                                .Where(tag => Title.IndexOf(tag, StringComparison.OrdinalIgnoreCase) == -1)
+                                // attributes concatenation, starting with Title
+                                .Aggregate(new StringBuilder(Title), (builder, attr) => builder.Append(" - ").Append(attr))
+                                .ToString();
+                        }
+
+                        return string.Join(" ", attributes);
                     }
 
-                    return string.Join(" ", attributes);
-                }
-
-                if (Type == MediaStreamType.Video)
-                {
-                    var attributes = new List<string>();
-
-                    var resolutionText = GetResolutionText();
-
-                    if (!string.IsNullOrEmpty(resolutionText))
+                    case MediaStreamType.Subtitle:
                     {
-                        attributes.Add(resolutionText);
+                        var attributes = new List<string>();
+
+                        if (!string.IsNullOrEmpty(Language))
+                        {
+                            // Get full language string i.e. eng -> English. Will not work for some languages which use ISO 639-2/B instead of /T codes.
+                            string fullLanguage = CultureInfo
+                                .GetCultures(CultureTypes.NeutralCultures)
+                                .FirstOrDefault(r => r.ThreeLetterISOLanguageName.Equals(Language, StringComparison.OrdinalIgnoreCase))
+                                ?.DisplayName;
+                            attributes.Add(StringHelper.FirstToUpper(fullLanguage ?? Language));
+                        }
+                        else
+                        {
+                            attributes.Add(string.IsNullOrEmpty(localizedUndefined) ? "Und" : localizedUndefined);
+                        }
+
+                        if (IsDefault)
+                        {
+                            attributes.Add(string.IsNullOrEmpty(localizedDefault) ? "Default" : localizedDefault);
+                        }
+
+                        if (IsForced)
+                        {
+                            attributes.Add(string.IsNullOrEmpty(localizedForced) ? "Forced" : localizedForced);
+                        }
+
+                        if (!string.IsNullOrEmpty(Title))
+                        {
+                            return attributes.AsEnumerable()
+                                // keep Tags that are not already in Title
+                                .Where(tag => Title.IndexOf(tag, StringComparison.OrdinalIgnoreCase) == -1)
+                                // attributes concatenation, starting with Title
+                                .Aggregate(new StringBuilder(Title), (builder, attr) => builder.Append(" - ").Append(attr))
+                                .ToString();
```

Comment:
```suggestion
                           var result = new StringBuilder(Title);
                            foreach (var tag in attributes)
                            {
                                // Keep Tags that are not already in Title.
                                if (Title.IndexOf(tag, StringComparison.OrdinalIgnoreCase) == -1)
                                {
                                    result.Append(" - ").Append(tag);
                                }
                            }

                            return result.ToString();
```

Prefer to not use Linq. Same for Movie and Subtitle